src/components/MermaidDash/components/LoadingIndicator.jsx

```diff
@@ -49,24 +49,25 @@ const StyledHeader = styled.header`
   `)}
 `
 
-const LoadingIndicator = ({ projectData, showLoadingIndicator, setShowLoadingIndicator }) => {
+const LoadingIndicator = ({ loadedProjectsCount, totalProjectsCount, isApiDataDoneLoading }) => {
   const [loadingProgressValue, setLoadingProgressValue] = useState(0)
+  const [showLoadingIndicator, setShowLoadingIndicator] = useState(!isApiDataDoneLoading)
 
   const _calculateCurrentLoadingPercentage = useEffect(() => {
-    if (!projectData.count || !projectData.results) {
-      return
+    if (totalProjectsCount === 0 || !totalProjectsCount) {
+      return // we cant divide by zero
     }
-    setLoadingProgressValue(Math.floor((projectData.results.length / projectData.count) * 100))
-  }, [projectData])
+    setLoadingProgressValue(Math.floor((loadedProjectsCount / totalProjectsCount) * 100))
+  }, [loadedProjectsCount, totalProjectsCount])
 
```

Comment:
Since we dont update the apiData to react state until its done loading now, we need something other simpler state to track progress. I didnt realize the rabbit hole I was creating until after I updated the fetching logic and noticed the progress indicator when from 0 to 100% with nothing in between.